src/Microsoft.Health.CosmosDb/Features/Storage/CollectionInitializer.cs

```diff
@@ -0,0 +1,55 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Health.CosmosDb.Features.Storage.Versioning;
+
+namespace Microsoft.Health.CosmosDb.Features.Storage
+{
+    public class CollectionInitializer : ICollectionInitializer
```

Comment:
The properties don't need to be publicly settable.

1. Make CollectionInitializer abstract (since this doesn't seem to be used directly).
2. Create a protected constructor that takes CollectionId, RelativeDatabaseUri, and etc.
3. Make the properties public getter and private setter.
4. Make the ControlPlaneCollectionInitialier and FhirCollectionInitializer call the protected constructor.

Reply:
Refactored the `CollectionInitializer` and got rid of the public properties since they aren't needed.